content/ADR-138-social-service-authentication.md

```diff
@@ -0,0 +1,101 @@
+---
+adr: 138
+date: 2022-11-28
+title: Social service authentication
+authors:
+  - jmoguilevsky
+status: Draft
+type: RFC
+spdx-license: CC0-1.0
+---
+
+## Abstract
+
+<!--
+Insert here a brief paragraph summarizing the RFC in its current state.
+This section shall provide an overview of whether this is a settled
+decision, alternatives explored and a short summary of relevant
+background information and key insights.
+-->
+
+As part of the social service initiative users will need to be able to authenticate themselves to be able to interact with other users and be able to access their information correctly.
+
+## Approach
+
+<!--
+How do you intend on addressing the need?  Describe what you plan on
+doing and the rationale behind the decisions you propose.  Then lay out
+the plan of execution, in rough order of how the execution should take
+place.  Include the rollout plan as well. (This is usually the longest
+section of the RFC) Hint: don’t be afraid of posting illustrations! The
+level of detail here has to be enough to give the reader  a clear
+understanding of the solution - it is up to the writer to decide.
+Further detail can be addressed to satisfy comments and increase clarity.
+-->
+
+The logging in system for the social needs to solve three problems:
+
+1. The user is correctly authenticated (the token is valid)
+1. Who the user is (which userId does the token belong to)
+1. Does the user have the correct permissions to perform a certain action (Authorization)
+
+The first two problems are already solved by the current chat solution (Matrix), so we propose to leverage the already existing implementation for simplicity and velocity in development. Matrix will be used as an authentication system, users will login against their service, then each endpoint will first validate against Matrix that the user is still using a valid token.
+
+The service will have a distributed cache (without persistance) with user-to-token information which will be used both to obtain the user (once the token is validated) and to prevent hitting Matrix for every request.
```

Comment:
Small suggestion:
```suggestion
The service will have a distributed cache (without persistence) with user-to-token information, which will be used to obtain the user (once the token is validated) and to prevent hitting Matrix for every request.
```